### PR TITLE
Fix Spark 3.4 build

### PR DESCRIPTION
**Pull request by razajafri**
This PR moves the `PartitionedFileUtils` to a shim to handle the introduction of `SparkPath` and adds `toString()` when calling `filePath` on a `PartitionedFile` object

fixes https://github.com/NVIDIA/spark-rapids/issues/7557.
fixes https://github.com/NVIDIA/spark-rapids/issues/7615

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

### CODE REVIEW

**Comment by razajafri** [comment]
build

**Comment by res-life** [comment]
Similar with this: https://github.com/NVIDIA/spark-rapids/pull/7635

**Review by jlowe — DISMISSED** [review]

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
@jlowe can you PTAL once more?

File: sql-plugin/src/test/330/scala/com/nvidia/spark/rapids/shims/spark330/Spark330ShimsSuite.scala [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021-2022, NVIDIA CORPORATION.
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/test/330/scala/com/nvidia/spark/rapids/shims/spark330/Spark330ShimsSuite.scala:2]: Copyright change but no other changes?

File: sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/PartitionedFileUtilsShim.scala [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, NVIDIA CORPORATION.
+ * Copyright (c) 2023, NVIDIA CORPORATION.
```
Comment (jlowe) [inline sql-plugin/src/main/311until340-all/scala/com/nvidia/spark/rapids/shims/PartitionedFileUtilsShim.scala:2]:
```suggestion
 * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```

**Review by jlowe — APPROVED** [review]

**Comment by razajafri** [comment]
build

**Pull request merged**